cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -19,6 +19,7 @@
 #include <cctype>
 #include <iterator>
 #include <string>
+#include <typeinfo>
```

Comment:
I don't think this is used, is it?

Reply:
No, I used it when trying to print the StringTransform type using `typeid(t).name()` but it printed more info than needed.